learn/features/randomness.md

```diff
@@ -0,0 +1,95 @@
+    | Block expiration delay |   {{ networks.moonbase.randomness.block_expiration }} blocks   |
+    | Epoch expiration delay |   {{ networks.moonbase.randomness.epoch_expiration }} epochs   |
+    |  Minimum block delay   | {{ networks.moonbase.randomness.min_vrf_blocks_delay }} blocks |
+    |  Maximum block delay   | {{ networks.moonbase.randomness.max_vrf_blocks_delay }} blocks |
+    |  Maximum random words  |   {{ networks.moonbase.randomness.max_random_words }} words    |
+
+## Local VRF {: #local-vrf }
+
+Local VRF randomness is generated on a block-by-block basis at the beginning of the block using the previous block's VRF output along with the public key of the current block author's VRF key. The generated randomness result is stored and used to fulfill all randomness requests for the current block.
+
+You can request local VRF randomness using the [`requestLocalVRFRandomWords` method](/builders/pallets-precompiles/precompiles/randomness/#:~:text=requestLocalVRFRandomWords){target=_blank} of the [randomness precompile](/builders/pallets-precompiles/precompiles/randomness/){target=_blank}.
+
+If your contract could have concurrent requests open, you can use the `requestId` returned from the `requestLocalVRFRandomWords` method to track which response is associated with which randomness request.
+
+## BABE Epoch Randomness {: #babe-epoch-randomness }
+
+BABE epoch randomness is based on a hash of the VRF values from the blocks produced in the relay chain epoch before last. On Polkadot, an [epoch lasts for roughly 4 hours](https://wiki.polkadot.network/docs/maintain-polkadot-parameters#periods-of-common-actions-and-attributes){target=_blank}, and on Kusama, an [epoch lasts for roughly 1 hour](https://guide.kusama.network/docs/kusama-parameters/#periods-of-common-actions-and-attributes){target=_blank}. The hashing is completed on the relay chain, and as such, it is not possible for a collator on Moonbeam to influence the randomness value unless they are also a validator on the relay chain and were responsible for producing the last output included in an epoch.
+
+The randomness is constant during a full epoch, so if a collator skips block production, the next collator can fulfill it with the same random value.
+
+You can request BABE epoch randomness using the [`requestRelayBabeEpochRandomWords` method](/builders/pallets-precompiles/precompiles/randomness/#:~:text=requestRelayBabeEpochRandomWords){target=_blank} of the [randomness precompile](/builders/pallets-precompiles/precompiles/randomness/){target=_blank}. In order to generate unique randomness, a different salt must be provided to the `requestRelayBabeEpochRandomWords` function.
+
+At the beginning of each relay chain epoch change, the randomness from one epoch ago is read from the relay chain state proof and used to fulfill all randomness requests that are due in the current block.
+
+## Request & Fulfill Process {: #request-and-fulfill-process }
```

Comment:
I would leave this decision to you, but this section seems more technical and not really worthy of being part of the **Learn** section right? It seems more precompile focused?

I think it is cool to leave a `Request & Fulfill Process` here but if it is more high-level oriented - But I'll leave it up to you to decide.

After I read the precompiles page, I noted that you refer this page there. So I'm ok with whatever you decide